### ISSUE TITLE
Logging exceptions doesn't include filename in the kibana apm dashboard

### ISSUE THREAD

**Issue by mterwoord** [elastic/apm-agent-dotnet#440]
**Describe the bug**
I have a simple asp.net core web application, which logs to my APM server. Now I have a controller action that does a simple ```throw new Exception("My error!");```
This exception is showing up in the apm dashboard in kibana, but the filename shown is the full class name of my controller, not the filename.

**To Reproduce**
1. Create a new asp.net core web application
2. Enable apm logging
3. Add the aforementioned exception statement to an action
4. Trigger the exception
5. See the exception in the kibana apm dashboard.

**Expected behavior**
The file location (line number) is correct, but the file name isn't.

**Comment by SergeyKleyman** [comment]
@mterwoord Thank for reporting the issue. We can confirm that it's a known issue (for example see #265) and we will update this issue soon on how we plan to handle it.

**Comment by mterwoord** [comment]
@SergeyKleyman Thanks for letting me know. I subscribed to the notifications of that issue, and hope the issue will be solved soon. Keep up the great work!

**Comment by SergeyKleyman** [comment]
@mterwoord if you don't mind I would like to keep this issue open since #265 is much more narrow in scope and #265 is not a bug issue.

**Comment by mterwoord** [comment]
If there is anything I can do to solve this issue, I'd be happy to help.

**Comment by SergeyKleyman** [comment]
Thank you very much for the offer - the solution is very simple but we need to discuss internally how to reconcile it with [the past decisions on which the current implementation is based](https://github.com/elastic/apm-agent-dotnet/pull/240#discussion_r289619196).

If you would like to offer a PR you are very welcome to do so.

**Comment by mterwoord** [comment]
So, basically first you guys need to decide internally first? Then I'll hold off on a PR for now, if you don't mind..

**Comment by gregkalapos** [comment]
This is somewhat of a duplicate of https://github.com/elastic/apm-agent-dotnet/issues/677.

A PR to clean this up is already open: https://github.com/elastic/apm-agent-dotnet/pull/848 

Once #848 is merged we'll send the file name in the `filename` field and the class name in the `classname` field, so you'll have both. Filename is either the `.cs` file name if we have pdbs, or the file name of the assembly that contains the given class. 

The reason we haven't sent the file name before was that there was no field to send the class name to the APM server, therefore we decided to put the classname into the `filename` field. The thinking behind was that class name in .NET is more important given that in a single file (`.cs` file or a big `.dll` assembly) you could have lots of classes.

But with #848 this'll be over and we'll do this nicely.
